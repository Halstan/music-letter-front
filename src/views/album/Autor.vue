<template>
  <b-container>
    <h2>Autor component</h2>
  </b-container>
</template>

<script>
import Autor from "../../models/autor.model";

export default {
  name: "AutorComponent",
  props: {
    autor: {
      required: true,
      type: Autor
    }
  }
};
</script>
