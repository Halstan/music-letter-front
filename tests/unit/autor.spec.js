import { createLocalVue, shallowMount } from "@vue/test-utils";
import Autor from "../../src/views/album/Autor.vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Autor.vue", () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });
  it("Renders correctly the autor information", () => {
    const autor = {
      idAutor: 1,
      nombres: "José",
      apellidos: "Madero Vizcaíno",
      urlFoto:
        "https://images.vice.com/vice/images/articles/meta/2016/12/02/entrevistamos-a-jose-madero-vocalista-de-pxndx-1480712778.jpg",
      fechaNacimiento: "1980-08-30",
      alias: "Pepe",
      biografia:
        "José Madero Vizcaíno, más conocido como Pepe Madero es un cantante, músico, compositor, productor, abogado, escritor y podcaster mexicano."
    };
    const wrapper = shallowMount(Autor, {
      propsData: { autor },
      vuetify,
      localVue
    });
    expect(wrapper.props("autor")).toBe(autor);
  });

  it("Checks if the v-card-subtitle contains the correct name & lastname", () => {
    const autor = {
      idAutor: 1,
      nombres: "José",
      apellidos: "Madero Vizcaíno",
      urlFoto:
        "https://images.vice.com/vice/images/articles/meta/2016/12/02/entrevistamos-a-jose-madero-vocalista-de-pxndx-1480712778.jpg",
      fechaNacimiento: "1980-08-30",
      alias: "Pepe",
      biografia:
        "José Madero Vizcaíno, más conocido como Pepe Madero es un cantante, músico, compositor, productor, abogado, escritor y podcaster mexicano."
    };
    const wrapper = shallowMount(Autor, {
      propsData: { autor },
      vuetify,
      localVue
    });
    expect(wrapper.find("div").text()).toContain("José");
  });
});
