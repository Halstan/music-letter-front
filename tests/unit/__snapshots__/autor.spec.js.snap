// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autor.vue Checks if the v-img contains the correct src of the autor 1`] = `
<div>
  <v-card-subtitle-stub>Autor: Jose Madero Vizcaino</v-card-subtitle-stub>
  <v-img-stub height="175" width="175" alt="Jose" options="[object Object]" position="center center" src="https://images.vice.com/vice/images/articles/meta/2016/12/02/entrevistamos-a-jose-madero-vocalista-de-pxndx-1480712778.jpg" transition="fade-transition"></v-img-stub>
</div>
`;
